compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -209,6 +324,9 @@ TR::Instruction *loadConstant(TR::CodeGenerator *cg, TR::Node * node, int64_t va
       return loadConstant(cg, node, (int32_t)value, trgReg, cursor, isPicSite);
       }
 
+   if (useTOC && cg->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_PPC_P10))
+      return loadAddressConstantInSnippet(cg, node, value, trgReg, NULL, TR::InstOpCode::ld, isPicSite, cursor);
```

Comment:
You're right that if a 32-bit machine reached this point, this would cause problems. However, 32-bit builds aren't supposed to be calling this helper in the first place, as it's meant to load a 64-bit constant into a single register; they're meant to call the version that uses a register pair instead.

Reply:
Might be helpful to put a comment in the code about that.

Reply:
Good point. Added an assert and comment to this function in 663c5c87681a4945e3d92bac420f6b1a45f1e806.